util/missingcrypto.txt

```diff
@@ -803,11 +808,11 @@ OCSP_REQUEST_get_ext_by_critical(3)
 OCSP_REQUEST_get_ext_count(3)
 OCSP_REQUEST_it(3)
 OCSP_REQUEST_print(3)
-OCSP_REQ_CTX_get0_mem_bio(3)
-OCSP_REQ_CTX_http(3)
-OCSP_REQ_CTX_nbio(3)
-OCSP_REQ_CTX_nbio_d2i(3)
-OCSP_REQ_CTX_new(3)
+OSSL_HTTP_REQ_CTX_get0_mem_bio(3)
+OSSL_HTTP_REQ_CTX_header(3)
+OSSL_HTTP_REQ_CTX_nbio(3)
+OSSL_HTTP_REQ_CTX_sendreq_d2i(3)
+OSSL_HTTP_REQ_CTX_new(3)
```

Comment:
I think we need to document these things before this PR goes in (or document them in this PR). We are introducing a new namespace (OSSL_HTTP) here. We should not instantly pollute it with undocumented functions. I realise that these were undocumented before -  but this increases the size of the problem. We used to have one set of functions that were undocumented - we now have a new set of functions plus a set of aliases which are all undocumented.

Reply:
>we need to document these things.

That's a reasonable attitude, but I am not going to do that. I think those involved with the original work should do so; they are free to take over this PR, or push commits, mail me files, whatever works.  (If I were paranoid, I would say that not wanting to do documentation was a motivation for the original define hacks in ```http_local.h``` :)

Reply:
> That's a reasonable attitude, but I am not going to do that. I think those involved with the original work should do so; they are free to take over this PR, or push commits, mail me files, whatever works.

I second this.

> (If I were paranoid, I would say that not wanting to do documentation was a motivation for the original define hacks in http_local.h :)

My motivation for those "`#define` hacks" was a different one: I did not dare to touch the pretty old `OCSP_` function entries in `libcrypto.num`, fearing that doing so might break binary compatibility of `libcrypto` with older versions.

Reply:
So @ddvo, it seems you wrote these functions (or someone at your company).  If that's correct, do you have any plans to get them documented?  And if I'm wrong, sorry, can you help figure out who did add them?

Reply:
I did not write these functions - they have been around for many years.

I'm pretty sure that all of them were added by Stephen Henson - see for instance
```
git blame 29f178bddfdbd11218fbcba0b8060297696968e3^ crypto/ocsp/ocsp_ht.c|grep OCSP_REQ_CTX_new
6f9076ff370 (Dr. Stephen Henson      2012-11-28 16:22:53 +0000  64) OCSP_REQ_CTX *OCSP_REQ_CTX_new(BIO *io, int maxline)
```

According to `git blame`, the only other person having significant impact on that source file is Matt.
So @mattcaswell you seem the natural candidate to fill in the documentation for those old functions.

Reply:
I think my "significant impact" amounts to having run the automated script "openssl-format-source" script over the entire codebase. So, by that measure, I think its my job to document *everything* :-)

Reply:
BTW, I still wonder why Stephen Henson left the project - when he did I could not find any information on that.

Reply:
> I think my "significant impact" amounts to having run the automated script "openssl-format-source" script over the entire codebase. So, by that measure, I think its my job to document everything :-)

Matt's commit 0f113f3ee4d is the great annotation wall. It's impossible to look through it, you need to climb over it if you want to blame other people.

Reply:
Since @levitte wrote the missing docs, I am resolving this issue.

Reply:
> Matt's commit 0f113f3 is the great annotation wall. It's impossible to look through it, you need to climb over it if you want to blame other people.

There is usually a way over, around, below, or through almost any wall :wink:
As far as analyzing the change history is concerned, as already mentioned above, there is a very simple way:
```
git blame 0f113f3^ crypto/ocsp/ocsp_ht.c
```

I'm unresolving this (part of the) issue just to make my comment to @mspncp better visible.